src/Countries/Bahrain.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Bahrain extends Country
+{
+    public function countryCode(): string
+    {
+        return 'bh';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        throw new \Exception('Not implemented yet.');
+
+        return array_merge([
+            'New Year Day' => '01-01',
+            'Eid al Fitr Holiday 1' => '04-10',
+            'Eid al Fitr Holiday 2' => '04-11',
+            'Eid al Fitr Holiday 3' => '04-12',
+            'Labour Day' => '05-01',
+            'Eid al Adha Holiday 1' => '06-16',
+            'Eid al Adha Holiday 2' => '06-17',
+            'Eid al Adha Holiday 3' => '06-18',
+            'Islamic New Year' => '07-07',
+            'Ashura Holiday 1' => '07-16',
+            'Ashura Holiday 2' => '07-17',
+            'Prophet Muhammads Birthday' => '07-15',
+            'National Day' => '12-16',
+            'National Day Holiday' => '12-17',
```

Comment:
```suggestion
            'New Year Day' => '01-01',
            'Labour Day' => '05-01',
            'National Day' => '12-16',
            'National Day Holiday' => '12-17',
```

These only the holidays that can be fixed, the removed ones cannot, because they are changing every year. Unless we have a dynamic way to calculate them.

Reply:
we update it every year

Reply:
It is difficult to calculate because the Eid depends on watching the moon
the best way is to have the date close. 
there have updates in commits example of calculation.